charts/argo-cd/templates/argocd-application-controller/statefulset.yaml

```diff
@@ -1,6 +1,6 @@
 {{- $redisHa := (index .Values "redis-ha") -}}
 apiVersion: apps/v1
-kind: Deployment
+kind: StatefulSet
```

Comment:
We've changed the way the StatefulSet will roll out with a feature flag until chart 3.x, could you please rebase this against master and fold the conditional in.

Reply:
Thx you for review. 
I think we can upgrade chart 3.0 for argocd 1.8, use StatefulSet for default.
> The argocd-application-controller converted to StatefulSet
in [1.7 to 1.8](https://argoproj.github.io/argo-cd/operator-manual/upgrading/1.7-1.8/ )

Reply:
Hi @Junnplus - I'd like this to be conditional for a while first. Please restore the conditional `kind` and `serviceName`.

Reply:
@seanson done.